src/test/java/com.teragrep.jos_01.procfs/LinuxOSTest.java

```diff
@@ -0,0 +1,212 @@
+                "fpu",
+                "fpu_exception",
+                "cpuid level",
+                "wp",
+                "flags",
+                "vmx flags",
+                "bugs",
+                "bogomips",
+                "clflush size",
+                "cache_alignment",
+                "address sizes",
+                "power management"
+        };
+
+        System.out.println(cpuinfo.statistics());
+        for (int i = 0; i < cpuinfo.cpuThreadCount(); i++) {
+            for (String key : expectedKeys) {
+                Assertions.assertTrue(cpuinfo.statistics().containsKey(key + "_" + i));
+            }
+        }
+    }
+
+    @Test
+    public void memInfoTest() {
+        LinuxOS os = new LinuxOS();
+        Meminfo meminfo = os.meminfo();
+        Assertions.assertEquals(56, meminfo.statistics().size());
+
+        String[] expectedKeys = {
+                "MemTotal",
+                "MemFree",
+                "MemAvailable",
+                "Buffers",
+                "Cached",
+                "SwapCached",
+                "Active",
+                "Inactive",
+                "Active(anon)",
+                "Inactive(anon)",
+                "Active(file)",
+                "Inactive(file)",
+                "Unevictable",
+                "Mlocked",
+                "SwapTotal",
+                "SwapFree",
+                "Zswap",
+                "Zswapped",
+                "Dirty",
+                "Writeback",
+                "AnonPages",
+                "Mapped",
+                "Shmem",
+                "KReclaimable",
+                "Slab",
+                "SReclaimable",
+                "SUnreclaim",
+                "KernelStack",
+                "PageTables",
+                "SecPageTables",
+                "NFS_Unstable",
+                "Bounce",
+                "WritebackTmp",
+                "CommitLimit",
+                "Committed_AS",
+                "VmallocTotal",
+                "VmallocUsed",
+                "VmallocChunk",
+                "Percpu",
+                "HardwareCorrupted",
+                "AnonHugePages",
+                "ShmemHugePages",
+                "ShmemPmdMapped",
+                "FileHugePages",
+                "FilePmdMapped",
+                "CmaTotal",
+                "CmaFree",
+                "HugePages_Total",
+                "HugePages_Free",
+                "HugePages_Rsvd",
+                "HugePages_Surp",
+                "Hugepagesize",
+                "Hugetlb",
+                "DirectMap4k",
+                "DirectMap2M",
+                "DirectMap1G"
+        };
+        System.out.println(meminfo.statistics());
+        for (String key : expectedKeys) {
+            Assertions.assertTrue(meminfo.statistics().containsKey(key));
+        }
+    }
+
+    @Test
+    public void highLevelMethodsTest() {
+        LinuxOS os = new LinuxOS();
+        Assertions.assertEquals(1, os.cpuCount());
```

Comment:
Will fail on non-single socket machines. This whole test will fail on other setups. Something like https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/lang/Runtime.html#availableProcessors-- could MAYBE be used as a workaround but I think it also contains thread count instead of logical core count.

Reply:
Used Runtime.availableProcessors() where applicable. Changed other tests to simply check that a positive integer gets returned, whatever the number of cpus the user's system ends up being